src/compiler/scala/tools/nsc/SubComponent.scala

```diff
@@ -14,7 +14,7 @@ package scala.tools.nsc
 
 import scala.ref.WeakReference
 
-/** An nsc sub-component.
+/** Compilation is split into phases and the sub-components of the compiler define such phases.
```

Comment:
It was actually an error to name it `SubComponent` instead of `Subcomponent`, as the word has no hyphen, except perhaps as variant. OED doesn't help me; subcommittee, subconscious; suburban; "sub-quality of humour" (of Chaucer). The code base has `ClassPath` and `ClassTag` but incorrectly `Classfile`; ch 4 of the jvm spec says "`class` File Format".

Reply:
Personally I think it should've just not had the "sub" part: there's are the components of the (whole) compiler.